rewards/utils/rewards_utils.py

```diff
@@ -78,6 +78,18 @@ def distribute_rewards_to_snapshot(
     return combine_rewards([rewards] + custom_rewards_list, 0)
 
 
+def distribute_rewards_from_total_snapshot(amount: int, snapshot: Snapshot, token: str):
+    rewards = RewardsList()
+    custom_rewards_list = []
+    total = snapshot.total_balance()
+    for addr, balance in snapshot:
+        rewards_percentage = Decimal(balance) / total if not total == 0 else 0
+        reward_amount = Decimal(amount) * rewards_percentage
+        assert reward_amount >= 0
+        rewards.increase_user_rewards(addr, token, reward_amount)
+    return combine_rewards([rewards] + custom_rewards_list, 0)
```

Comment:
Another magic number here - Not sure best practice but maybe we make a `combine_multiple_snapshot_rewards` wrapper function which only accepts `rewards_list: List[RewardsList]` and defaults cycle to 0?

```
def combine_multiple_snapshot_rewards(rewards_list: List[RewardsList]) -> RewardsList:
    return combine_rewards(rewards_list, 0)
```

Reply:
I removed magic number from here, but maybe let's consider refactoring this after tests and release. I might have some more time to design this better